File: docker/_solution/Dockerfile.c
FROM ubuntu

RUN apt update && apt install -y git gcc make libpcap-dev \

RUN git clone https://github.com/AstroTech/ecosystem-example-c /tmp \
    && cd /tmp \
    && ./configure \
    && make \
    && make install


## -----


FROM alpine

RUN apk add --no-cache git gcc musl-dev libpcap-dev make

RUN git clone https://github.com/AstroTech/ecosystem-example-c /tmp \
    && cd /tmp \
    && ./configure \
    && make \
    && make install

File: docker/_solution/Dockerfile.java
FROM alpine

RUN apk add git maven openjdk8
RUN git clone https://github.com/sages-pl/src-java /tmp

WORKDIR /tmp

# To run the unit tests, call mvn test
# To run the integration tests as well, call mvn verify
CMD mvn compile && mvn test && mvn verify
